source/common/tracing/http_tracer_impl.cc

```diff
@@ -163,6 +166,19 @@ void HttpTracerUtility::finalizeSpan(Span& span, const Http::HeaderMap* request_
   span.setTag(Tracing::Tags::get().ResponseFlags,
               StreamInfo::ResponseFlagUtils::toShortString(stream_info));
 
+  // GRPC data.
+  if (response_headers && response_trailers && stream_info.protocol() == Http::Protocol::Http2 &&
+      Grpc::Common::hasGrpcContentType(*response_headers)) {
+    const Http::HeaderEntry* grpc_status_header = response_trailers->GrpcStatus();
+    if (grpc_status_header) {
+      span.setTag(Tracing::Tags::get().GrpcStatusCode, grpc_status_header->value().getStringView());
+    }
+    const Http::HeaderEntry* grpc_message_header = response_trailers->GrpcMessage();
+    if (grpc_message_header) {
+      span.setTag(Tracing::Tags::get().GrpcMessage, grpc_message_header->value().getStringView());
+    }
+  }
```

Comment:
Would be good to also set the `error` tag, as for the [async client](https://github.com/envoyproxy/envoy/blob/master/source/common/grpc/async_client_impl.cc#L257-L261), if the status is not `Ok`.

Reply:
Thanks @objectiser - I've added this.